components/ai_chat/core/browser/constants.h

```diff
@@ -24,12 +24,20 @@ extern const base::fixed_flat_set<std::string_view, 1>
 inline constexpr uint8_t kMaxPreviewPages = 20;
 extern const char kLeoModelSupportUrl[];
 
-// Llama 3 has a max context length of 8k tokens. Phi 3 Mini and Llama 2 have 4k
-// tokens max. Using Phi 3 Mini as the upper bound, 1 token ≈ 4 characters, so
-// the max context length is ~16k characters. To allow for follow-ups, set max
-// page content length to 60% of this, i.e., 9.6k characters.
-inline constexpr int kCustomModelMaxPageContentLength = 9600;
-inline constexpr int kCustomModelLongConversationCharLimit = 10000;
+/**
+ * Upon registering a custom model, users have the ability to explicitly
+ * provide a context size (in tokens). When present, we'll use this value
+ * to determine the max page content length (in chars). We will assume 4
+ * chars per token. When no context size has been provided, we will
+ * default to a conservative 4k tokens based on common models like Phi 3
+ * Mini and Llama 2 (both have 4k token context limits).
+ */
+inline constexpr size_t kDefaultCharsPerToken = 4;
+inline constexpr float kMaxContentLengthThreshold = 0.6f;
+inline constexpr size_t kCustomModelDefaultContextSize = 4000;
+inline constexpr size_t kCustomMaxPageContentLength =
+    kCustomModelDefaultContextSize * kDefaultCharsPerToken *
+    kMaxContentLengthThreshold;
```

Comment:
cc @jonathansampson probably worth taking a look at this one, and sticking with either signed or unsigned numbers :smile:

Reply:
@fallaciousreasoning Currently walking through this now; I don't think I used any signed integers anywhere, but double-checking.

Reply:
@jonathansampson you are multiplying a size_t by a float (which is signed) and then casting back to a size_t (which is an unsigned int). The issue is not just signed/unsigned, you could also potentially overflow `kCustomMaxPageContentLength`. Where is `kCustomMaxPageContentLength` actually used? Could be github being stupid, but I don't see anything when searching for it.

Reply:
This is also doing implicit casts (bad) on top of the sketchy uint/float math

Reply:
It's actually not being used, @bridiver. We had a hard-coded value originally, IIRC. When I started moving to add support for custom context sizes (which impacts max-page-content-length, etc.,), I switched to doing the math instead. But ultimately that value (kCustomMaxPageContentLength) is not directly used anywhere (and will be removed).

Reply:
also why are you using a `size_t` here when you're using `uint32_t` in other places? Since you're always assigning the `size_t` to the `uint32_t` it will never overflow, but it's a bit confusing